tests/subset.rs

```diff
@@ -43,23 +43,26 @@ fn test_subset<A: Adversary<Subset<NodeId>>>(
         network.step();
     }
     // Verify that all instances output the same set.
-    let mut expected = None;
     for node in network.nodes.values() {
-        if let Some(output) = expected.as_ref() {
-            assert!(once(output).eq(node.outputs()));
-            continue;
+        let outputs = node.outputs();
+        let mut actual = BTreeMap::default();
+
+        for i in outputs {
+            match i {
+                SubsetOutput::Contribution(k, v) => {
+                    assert!(actual.insert(k, v).is_none());
+                }
+                SubsetOutput::Done => break,
+            }
+        }
+        assert_eq!(outputs.len(), actual.len() + 1);
+
+        // The Subset algorithm guarantees that more than two thirds of the proposed elements
+        // are in the set.
+        assert!(actual.len() * 3 > inputs.len() * 2);
+        for (id, value) in actual {
+            assert_eq!(&inputs[&id], value);
         }
-        assert_eq!(1, node.outputs().len());
-        expected = Some(node.outputs()[0].clone());
-    }
-    let output = expected.unwrap();
-    assert!(once(&output).eq(network.observer.outputs()));
```

Comment:
Can you please add back the test of outputs from the observer node? It looks that you've forgotten to add it back while refactoring.

Reply:
Done